cli/src/diff_util.rs

```diff
@@ -804,11 +808,18 @@ pub fn show_diff_stat(
     formatter.with_label("diff", |formatter| {
         let mut total_added = 0;
         let mut total_removed = 0;
-        for stat in &stats {
+        let total_files = stats.len();
+        for mut stat in stats {
             total_added += stat.added;
             total_removed += stat.removed;
             let bar_added = (stat.added as f64 * factor).ceil() as usize;
             let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+            if stat.path.len() > max_path_length {
+                stat.path = format!(
+                    "...{}",
+                    &stat.path[(stat.path.len() - 1 - (max_path_length - 3))..]
```

Comment:
`str[..]` panics if the index is not at char boundary. `std::fmt` appears to use `.char_indices().nth()` to truncate string. Maybe we can do that from right?

I also think `- 1` is wrong since the length including ellipsis shouldn't exceed the `max_path_length`.